### PR TITLE
Relocate release process documentation

### LINKED ISSUE #2958: Missing Release Process page in TOC
## Description

`/docs/latest/contribute/release-process/ ` is being indexed in Algolia, and appears as a search result, because it is listed in `sitemap.xml`. The page is not in the TOC or linked in other pages as far as I can tell.
- Should the page be listed on the site?
- and is the content still relevant? Last major update was in July, 2021

## Location

- https://tanzucommunityedition.io/docs/latest/contribute/release-process/
- https://github.com/vmware-tanzu/community-edition/blob/main/docs/site/content/docs/latest/contribute/release-process.md

### PR DESCRIPTION
**Pull request by davidvonthenen**
## What this PR does / why we need it

This just moves the release process documentation to a different folder in the repo and out of the user documentation. Also, removes pictures that no longer exist in the repo.

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
NONE
```

## Which issue(s) this PR fixes

Fixes: https://github.com/vmware-tanzu/community-edition/issues/2958

## Describe testing done for PR

NA

## Special notes for your reviewer

NA

### CODE REVIEW

**Comment by stmcginnis** [comment]
Is `docs/designs` the best place for this? I think it might be better to create a `docs/internal` or `docs/developer` or something rather than mixing this in with our design docs.

**Comment by dvonthenen** [comment]
> Is `docs/designs` the best place for this? I think it might be better to create a `docs/internal` or `docs/developer` or something rather than mixing this in with our design docs.

Will create a developer folder

File: docs/developer/release-process.md [hunk 1]
```diff
@@ -107,8 +107,6 @@ We are currently consuming the CLI from the Core repo using a Tanzu Community Ed
 
 The side effect or consequence for temporarily needing to rebuild all components is that it happens to also be useful for developers to build their own Tanzu Community Edition release locally for testing out features they would like to introduce or contribute. This can simply be done by running `make release`.
 
-![makefile](/docs/img/makerelease.png)
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply docs/developer/release-process.md:110]: I think those images were actually deleted when the docs website went live on the great doc branch merge. I could go back in history to look for them, but to be honest, they aren't too terribly relevant. It was more or less to have some pictures in the doc. 😀

[the remark this answers is not in the document]
Comment (dvonthenen) [reply docs/developer/release-process.md:110]: Nevermind... I found them. They still exist. Going to move them since they are there.

**Review by stmcginnis — APPROVED** [review]
Looks good!

**Pull request merged**